### DISCUSSION TITLE
How to create a custom hook with inferred return types

### DISCUSSION THREAD

**Discussion by morleytatro** [trpc/trpc#6171] [Q&A] [answered]
I'm looking to do the following:
https://stackoverflow.com/questions/77096846/wrap-usequery-of-trpc-in-a-custom-hook-with-correct-typescript-types

The solution mentioned works for v10 but the typing completely falls apart for v11. Looks like there's substantial interest from the community:
https://discord.com/channels/867764511159091230/1020132255567454310/threads/1151511450859356170

Has anyone made any headway on this or have any suggestions. @KATT?

**Comment by KATT** [comment]
We've simplified the router outputs to help with big project's TS perf this is a slight breaking change for you.

Your best bet is to make a feature request to supply documentation to achieve this. Might be that we need to supply some new helpers to achieve it.

**Reply by KATT** [reply, marked as answer]
See #6180 

Feel free to add a bounty
